cx/spark/server.js

```diff
@@ -0,0 +1,210 @@
+ */
+async function detectIntentText(message) {
+  const sessionId = message.payload.personId;
+  const sessionPath = client.projectLocationAgentSessionPath(
+      projectId,
+      locationId,
+      agentId,
+      sessionId
+  );
+  console.info(sessionPath);
+
+  const request = sparkToDetectIntent(message, sessionPath);
+  const [response] = await client.detectIntent(request);
+
+  return response;
+}
+
+function sendMessage(message) {
+  request.post('https://api.ciscospark.com/v1/messages', {
+    auth: {
+      bearer: sparkAccessToken
+    },
+    json: message
+  }, (err, resp, body) => {
+    if (err) {
+      console.error('Failed to send message :' + err);
+    }
+  });
+}
+
+function registerWebhook() {
+  request.post('https://api.ciscospark.com/v1/webhooks', {
+    auth: {
+      bearer: sparkAccessToken
+    },
+    json: {
+      "name": "test",
+      "targetUrl": targetUrl,
+      "resource": "messages",
+      "event": "created"
+    }
+  }, (err, resp, body) => {
+    if (err) {
+      console.error('Failed to create Webhook :' + err);
+    }
+  });
+}
+
+function deleteWebhooks() {
+  return new Promise((resolve, reject) =>{
+    request.get('https://api.ciscospark.com/v1/webhooks?max=100', {
+      auth: {
+        bearer: sparkAccessToken
+      }
+    }, (err, resp, body) => {
+      if (err) {
+        console.error('Failed to check webhooks :' + err);
+        reject();
+      }
+      var webhooks = JSON.parse(resp.body).items;
+      if (Array.isArray(webhooks)) {
+        webhooks = webhooks.filter((value, index, arr)=> {
+          return value.targetUrl===targetUrl;
+        });
+        webhooks.forEach((webhook) => {
+          request.delete(
+              'https://api.ciscospark.com/v1/webhooks/' +
+              webhook.id, {
+                auth: {
+                  bearer: sparkAccessToken
+                }
+              }, (err, resp, body) => {
+                if (err) {
+                  console.error('Failed to delete webhook :' + err);
+                }
+              });
+        });
+      }
+      resolve();
+    });
+  });
+}
+
+function retrieveMessage(messageId) {
+  return new Promise((resolve, reject) =>{
+    request.get('https://api.ciscospark.com/v1/messages/' + messageId, {
+      auth: {
+        bearer: sparkAccessToken
+      }
+    }, (err, resp, body) => {
+      if (err) {
+        console.error('Failed to retrieve message :' + err);
+        reject();
+      }
+      // checks to make sure the message is not from itself
+      if (!((JSON.parse(resp.body).personEmail).includes('webex.bot'))) {
+        const personEmail = JSON.parse(resp.body).personEmail;
+        const messageText= JSON.parse(resp.body).text;
+        const payload = JSON.parse(resp.body);
+        resolve({text: messageText, email: personEmail, payload:payload});
+      } else {
+        resolve(null);
+      }
+    });
+  });
+}
+
+  module.exports = {sparkToDetectIntent, detectIntentToSparkMessage};
```

Comment:
nit: wrong indent?

Could we try to use `clang-format` to make sure the code/tests are following Google style? So that we don't need to fix the style manually one by one.
E.g., https://github.com/google/closure-library/wiki/Formatting-.js-with-clang-format